schema_based_on_diagram.sql

```diff
@@ -0,0 +1,76 @@
+-- CREATED A NEW TABLE 'patients'
+CREATE TABLE patients(
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    name VARCHAR (250),
+    date_of_birth DATE
+);
+
+-- CREATED A NEW TABLE 'invoice'
+CREATE TABLE invoices(
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    total_amount decimal,
+    generated_at timestamp,
+    payed_at timestamp,
+    medical_history_id INT,
+    CONSTRAINT fk_medical_history
+    FOREIGN KEY (medical_history_id)
+    REFERENCES medical_histories (id),
+);
+
+-- CREATED A NEW TABLE 'medical_histories'
+CREATE TABLE medical_histories(
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    admitted_at timestamp,
+    patient_id int,
+    status VARCHAR(255),
+    CONSTRAINT fk_patient
+    FOREIGN KEY (patient_id)
+    REFERENCES patients (id)
+);
+
+-- CREATED A NEW TABLE 'invoices_items'
+CREATE TABLE invoices_items(
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    unit_price decimal,
+    quantity int,
+    total_price decimal,
+    invoice_id int,
+    treatment_id int,
+    CONSTRAINT fk_invoice
+    FOREIGN KEY (invoice_id)
+    REFERENCES invoices (id),
+    CONSTRAINT fk_treatment,
+    FOREIGN KEY (treatment_id)
+    REFERENCES treatments (id),
+);
+
+-- CREATED A NEW TABLE 'treatments'
+CREATE TABLE treatments(
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    type VARCHAR(255),
+    name VARCHAR(255),
+); 
+
+-- CREATED A NEW TABLE 'medical_histories_treatments'
+CREATE TABLE medical_histories_treatments(
+    medical_histories_id int,
+    treatments_id int,
+    FOREIGN KEY (medical_histories_id)
+    REFERENCES medical_histories (id),
+    FOREIGN KEY (treatments_id)
+    REFERENCES treatments (id),
+);
+
```

Comment:
- [ ] It would be better if we create a table between medical_histories and treatments. Currently, we do not have a table like that

Reply:
Actually we have the table as a medical_histories_treatments

Reply:
yeah, you are rignt. My bad. Kindly re-submit now so it get approved immediately okay. How did i miss that?

Reply:
Thank you so much